Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbApplication.cpp

```diff
@@ -34,6 +36,31 @@ namespace AzFramework
             return m_xcbConnection.get();
         }
 
+        void SetEnableXInput(xcb_connection_t* connection, bool enable) override
+        {
+            struct Mask
+            {
+                xcb_input_event_mask_t head;
+                xcb_input_xi_event_mask_t mask;
+            };
+            const Mask mask {
+                /*.head=*/{
+                    /*.device_id=*/XCB_INPUT_DEVICE_ALL_MASTER,
+                    /*.mask_len=*/1
+                },
+                /*.mask=*/ enable ?
+                    (xcb_input_xi_event_mask_t)(XCB_INPUT_XI_EVENT_MASK_RAW_MOTION | XCB_INPUT_XI_EVENT_MASK_RAW_BUTTON_PRESS | XCB_INPUT_XI_EVENT_MASK_RAW_BUTTON_RELEASE) :
+                    (xcb_input_xi_event_mask_t)XCB_NONE
+            };
+
+            const xcb_setup_t* xcbSetup = xcb_get_setup(connection);
+            const xcb_screen_t* xcbScreen = xcb_setup_roots_iterator(xcbSetup).data;
+
+            xcb_input_xi_select_events(connection, xcbScreen->root, 1, &mask.head);
```

Comment:
isnt `&mask.head` the same as `&mask`?

Reply:
The address is the same, but the type is not.

This function takes a list of ... stuff. Internally, it just typecasts that `const xcb_input_event_mask_t*` parameter to a `char*`. The xcb code is all autogenerated based on some X protocol definition, and I'm not familiar enough with the X protocol to grok exactly what it is doing. From what I can tell, it just needs a contiguous block of memory to read from. We accomplish that here by making a `struct`. We're kind of cheating, because the types are not the same:
* first member is a `xcb_input_event_mask_t` (a struct itself with 2 16-byte members),
* second member is a `xcb_input_xi_event_mask_t` (an `enum`, 32-bytes)
But everything works out because the `sizeof` matches (which I'm guessing is an intentional aspect of the X protocol).

However, to make the type system happy, we have to pass a `const xcb_input_event_mask_t*`, not a `const OurCustomMaskStruct*`.

Reply:
Got it, nice work-around